data_structures/heap/heap.py

```diff
@@ -1,99 +1,131 @@
-#!/usr/bin/python3
+from typing import Iterable, List, Union
 
 
 class Heap:
-    """
+    """A Max Heap Implementation
+
     >>> unsorted = [103, 9, 1, 7, 11, 15, 25, 201, 209, 107, 5]
     >>> h = Heap()
-    >>> h.build_heap(unsorted)
-    >>> h.display()
+    >>> h.build_max_heap(unsorted)
+    >>> print(h)
     [209, 201, 25, 103, 107, 15, 1, 9, 7, 11, 5]
     >>>
-    >>> h.get_max()
+    >>> h.extract_max()
     209
-    >>> h.display()
+    >>> print(h)
     [201, 107, 25, 103, 11, 15, 1, 9, 7, 5]
     >>>
     >>> h.insert(100)
-    >>> h.display()
+    >>> print(h)
     [201, 107, 25, 103, 100, 15, 1, 9, 7, 5, 11]
     >>>
     >>> h.heap_sort()
-    >>> h.display()
+    >>> print(h)
     [1, 5, 7, 9, 11, 15, 25, 100, 103, 107, 201]
-    >>>
     """
 
     def __init__(self):
-        self.h = []
-        self.curr_size = 0
+        self.h: List[Union[int, float]] = []
```

Comment:
`int` and `float` have a subtype relationship and in a union, only the less specific type survives, which means:
```suggestion
        self.h: List[float] = []
```

Reply:
got it, nice to know!